roadmap/implementers-guide/src/node/approval/approval-voting.md

```diff
@@ -0,0 +1,247 @@
+# Approval Voting
+
+Reading the [section on the approval protocol](../../protocol-approval.md) will likely be necessary to understand the aims of this subsystem.
+
+## Protocol
+
+Input:
+  - `ApprovalVotingMessage::CheckAndImportAssignment`
+  - `ApprovalVotingMessage::CheckAndImportApproval`
+  - `ApprovalVotingMessage::ApprovedAncestor`
+
+Output:
+  - `ApprovalNetworkingMessage::DistributeAssignment`
+  - `ApprovalNetworkingMessage::DistributeApproval`
+  - `RuntimeApiMessage::Request`
+  - `ChainApiMessage`
+  - `AvailabilityRecoveryMessage::Recover`
+  - `CandidateExecutionMessage::ValidateFromExhaustive`
+
+## Functionality
+
+The approval voting subsystem is responsible for casting votes and determining approval of candidates and as a result, blocks.
+
+This subsystem wraps a database which is used to store metadata about unfinalized blocks and the candidates within them. Candidates may appear in multiple blocks, and assignment criteria are chosen differently based on the hash of the block they appear in.
+
+## Database Schema
+
+The database schema is designed with the following goals in mind:
+  1. To provide an easy index from unfinalized blocks to candidates
+  1. To provide a lookup from candidate hash to approval status
+  1. To be easy to clear on start-up. What has happened while we were offline is unimportant.
+  1. To be fast to clear entries outdated by finality
```

Comment:
It's maybe implicit when you talk about the unfinalized relay chain block here, but I needed maps between paraids and cores-when-included aka when declared available in preliminary draft: https://github.com/paritytech/polkadot/pull/1558/files#diff-e9db7842138e4616e3dd5824d7df2f9947e2dcc7570c5fc20d755ec6947ab98fR97-R112

Reply:
We use `CandidateEvents` for that.